piControlMain.c

```diff
@@ -199,21 +199,39 @@ static int __init piControlInit(void)
 
 	if (of_machine_is_compatible("kunbus,revpi-compact")) {
 		piDev_g.machine_type = REVPI_COMPACT;
+		piDev_g.pibridge_rs485_supported = 0;
+		piDev_g.pibridge_ethernet_p2p_supported = 0;
+		piDev_g.pibridge_ethernet_supported = 0;
```

Comment:
Global variables are <a href="https://stackoverflow.com/a/20916572">initialized to zero</a> in C. Per convention, members of a global struct aren't additionally initialized to zero at runtime. So you only need to add lines here which set a feature flag to 1 and you can drop all the other ones which set a flag to zero.